### PR TITLE
New RDataFrame Tutorials

### PR DESCRIPTION
**Pull request by AlexandraDobrescu**
The tutorials demonstrate how RDataFrame can be used to create a connection with a SQlite3 database.

### CODE REVIEW

**Comment by bluehood** [comment]
hi, RSqliteDS is not in ROOT yet, so the PR build will fail, right?

**Comment by bluehood** [comment]
(will start a build when the datasource is in)

**Comment by couet** [comment]
We wait Jakob to commit this one.

File: tutorials/dataframe/df026_SQliteIPLocation.C [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   WM = (TH2Poly*) F->Get("WMUSA");
+
+   auto fillIPLocation = [&WM] ( const std::string &sLongitude, const std::string &sLatitude ) {
+      if (!( sLongitude == "" ) && !( sLatitude == "" )) {
+         float latitude = std::stof(sLatitude);
+         float longitude = std::stof(sLongitude);
+         WM->Fill(longitude, latitude);
+      }
+   };
+
```
Comment (dpiparo) [inline tutorials/dataframe/df026_SQliteIPLocation.C]: Can we use 2 defines, convert to numbers and use the fill action of rdf?

File: tutorials/dataframe/df025_SQliteDependencyOverVersion.C [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   auto rdf = ROOT::RDF::MakeSqliteDataFrame( "https://root.cern.ch/download/root_download_stats.sqlite", "SELECT * FROM accesslog;" );
+
+   auto minTime = *rdf.Reduce([](std::string a, std::string b) {return std::min(a, b);}, "Time", std::string("Z"));
+   std::cout << "Minimum time is '" << minTime << "'" << std::endl;
+   TDatime minTimeFormat(minTime.c_str());
+
+   TDatime now;
+   TH1F hTime( "hTime", "Duration of ROOT dependency over version 6.14", 10, minTimeFormat.Convert(), now.Convert() );
+
+   auto fillTimeHisto = [&hTime] ( const std::string &time, const std::string &version ) {
```
Comment (dpiparo) [inline tutorials/dataframe/df025_SQliteDependencyOverVersion.C]: but do we need the foreach? Why not converting these strings to numbers with a define and then fill the histogram?

File: tutorials/dataframe/df026_SQliteIPLocation.C [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   rdf.Foreach( fillIPLocation, { "IPLongitude", "IPLatitude" } );
+
+   TCanvas *locationHistogram = new TCanvas();
+
+   locationHistogram->SetLogz(1);
+   locationHistogram->ToggleEventStatus();
```
Comment (dpiparo) [inline tutorials/dataframe/df026_SQliteIPLocation.C]: Why this?

File: tutorials/dataframe/df027_SQlitePlatformDistribution.C [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+/// \macro_image
+///
+/// \author Alexandra-Maria Dobrescu 08/2018
+
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RDataSource.hxx"
+#include "ROOT/RSqliteDS.hxx"
+#include <TString.h>
+#include <sqlite3.h>
+
```
Comment (dpiparo) [inline tutorials/dataframe/df027_SQlitePlatformDistribution.C]: Headers.

File: tutorials/dataframe/df027_SQlitePlatformDistribution.C [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        hShortRootPlatform.Fill(Platform_0,1);
+      } else if ( Platform.Contains("Linux") ){
+        hShortRootPlatform.Fill(Platform_0,1);
+      } else if ( Platform.Contains("source") ){
+        hShortRootPlatform.Fill(Platform_1,1);
+      } else if ( Platform.Contains("macosx64") ){
+        hShortRootPlatform.Fill(Platform_2,1);
+      } else if ( Platform.Contains("IRIX64") ){
+        hShortRootPlatform.Fill(Platform_1,1);
+      }
```
Comment (dpiparo) [inline tutorials/dataframe/df027_SQlitePlatformDistribution.C]: Can we do this with Filters?

File: tutorials/dataframe/df027_SQlitePlatformDistribution.C [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+   rdf.Foreach( fillRootPlatform, { "Platform" } );
+
+   TCanvas *PlatformDistributionHistogram = new TCanvas();
+
+   hRootPlatform.GetXaxis()->LabelsOption("a");
+   hRootPlatform.LabelsDeflate("X");
+   hRootPlatform.DrawClone();
+
+   TCanvas *shortPlatformDistributionHistogram = new TCanvas();
```
Comment (dpiparo) [inline tutorials/dataframe/df027_SQlitePlatformDistribution.C]: auto

File: tutorials/dataframe/df028_SQliteVersionsOfROOT.C [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+/// \macro_image
+///
+/// \author Alexandra-Maria Dobrescu 08/2018
+
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RDataSource.hxx"
+#include "ROOT/RSqliteDS.hxx"
+#include <TString.h>
+#include <sqlite3.h>
+
```
Comment (dpiparo) [inline tutorials/dataframe/df028_SQliteVersionsOfROOT.C]: no headers.

File: tutorials/dataframe/df028_SQliteVersionsOfROOT.C [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+   TH1F hVersionOfRoot("hVersionOfRoot", "Development Versions of ROOT", 7, 0, -1);
+
+   auto fillVersionHisto = [&hVersionOfRoot] (const std::string &version) {
+      TString copyVersion = version;
+      TString shortVersion(copyVersion(0,4));
+      hVersionOfRoot.Fill(shortVersion, 1);
+   };
+
+   rdf.Foreach( fillVersionHisto, { "Version" } );
```
Comment (dpiparo) [inline tutorials/dataframe/df028_SQliteVersionsOfROOT.C]: Again: filters and defines?

**Comment by bluehood** [comment]
@phsft-bot build

**Comment by dpiparo** [comment]
they do not @bluehood

**Comment by bluehood** [comment]
@dpiparo so there is no point in asking @phsft-bot to run...too bad :sweat_smile:

**Comment by Axel-Naumann** [comment]
@dpiparo @couet could you coordinate to get this into master?

**Comment by dpiparo** [comment]
reviewed in August, waiting for olivier to review.

**Comment by dpiparo** [comment]
The pr has been integrated w/o waiting any more for the scheduled reviews.

**Pull request closed without merging**